test/src/fakes.dart

```diff
@@ -70,7 +70,8 @@ Future<Map<String, dynamic>> _getJsonRequestBody(
 ) async {
   final requestBytesStream = request.finalize();
   final requestBodyString = await requestBytesStream.bytesToString();
-  if (requestBodyString.isEmpty) {
+  // Multipart request body (e.g. when uploading a file) is not a JSON string
+  if (requestBodyString.isEmpty || !requestBodyString.startsWith('{')) {
```

Comment:
Looks ok but what if it's json array? Probably we should also check for `[`. Or maybe there is some better way to detect json. Wdyt?

Reply:
Good point 👍 Since the multipart request body starts with the delimiter (a few `-` followed by a random string, e.g. "--2a8ae6ad-f4ad-4d9a-a92c-6d217011fe0f") I check if the body starts with `-` and if it does I assume it's not a JSON 😉 https://github.com/nomtek/mistralai_client_dart/pull/59/commits/33060093fc6f5b18a41e1218b7455bab70096f2c